### PR TITLE
Uapi additions

### PR DESCRIPTION
**Pull request by kellygold**
added metadata APIs, integrated passthrough requests into each Commerce, Accounting, and CRM categories. Added Test files and Docs for all of these

### CODE REVIEW

File: alloy_python/uapi/crm.py [hunk 1]
```diff
@@ -168,3 +168,20 @@ def update_user(self, user_id, data):
...
+            "method": method,
+            "endpoint": endpoint,
+            "body": body,
+            "query": query,
+            "extraHeaders": extra_headers
+        }
+        try:
+            response = requests.post(passthrough_url, headers=self.headers, json=payload)
+            response.raise_for_status()
+            return response.json()
```
Comment (greggmojica) [inline alloy_python/uapi/crm.py]: Let's add a bit more error handling

File: alloy_python/uapi/metadata.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    def _api_request(self, method, url, params=None):
+        try:
+            response = requests.request(
+                method,
+                url,
+                headers=self.headers,
+                params=params
+            )
+            response.raise_for_status()
```
Comment (greggmojica) [inline alloy_python/uapi/metadata.py:28]: Same here

File: test/uapi_passthrough/commerce.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+commerce = Commerce(api_key)
+commerce.connect(connection_id)
+
+# Example passthrough request to an external API endpoint
+response = commerce.passthrough_request(
+    method="GET",
+    endpoint="/customers.json",  # Replace with a valid endpoint
+    query={"limit": 10}
+)
```
Comment (greggmojica) [inline test/uapi_passthrough/commerce.py]: Break passthrough into its own class

Comment (kellygold) [reply test/uapi_passthrough/commerce.py]: Done

File: alloy_python/uapi/metadata.py (hunk 2 above)
Comment (kellygold) [reply alloy_python/uapi/metadata.py:28]: Updated classes to align with existing error handling for CRM, Accounting, Commerce. 

The raise_for_status() is a trigger for the except blocks, which have the acutual error handling logic. 

Currently, these return the response body JSON as received from the API (error message) if there's any error
image

**Comment by greggmojica** [comment]
Approved

**Review by greggmojica — APPROVED** [review]

**Pull request merged**